backend/src/main/java/com/github/mysterix5/vover/user_details/VoverUserDetailsService.java

```diff
@@ -14,18 +16,33 @@
 @RequiredArgsConstructor
 public class VoverUserDetailsService {
     private final VoverUserDetailsMongoRepository userDetailsRepository;
+    private final HistoryService historyService;
+
+    public void addRequestToHistory(String username, List<RecordDbEntity> recordList) {
```

Comment:
It might be a good idea to implement the `HistoryService` as an observer.

Reply:
yes i write that on my list ;)